src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Model/SourceGenerationSpec.cs

```diff
@@ -1,21 +1,23 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Collections.Generic;
 
 namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
 {
     internal sealed record SourceGenerationSpec
     {
+        public Dictionary<Enum, List<InterceptorLocationInfo>> GenMethodsInterceptionInfo { get; } = new();
         public Dictionary<MethodsToGen_CoreBindingHelper, HashSet<TypeSpec>> TypesForGen_CoreBindingHelper_Methods { get; } = new();
-        public Dictionary<MethodsToGen_ConfigurationBinder, HashSet<TypeSpec>> TypesForGen_ConfigurationBinder_BindMethods { get; } = new();
```

Comment:
We don't need to generate typed interceptors for each `ConfigurationBinder.Bind` invocation anymore. Instead, we generate one interceptor for each overload that's called, and funnel into one `BindCoreMain` method with type checks.

Reply:
This is worth adding it as a comment in the code. It is really helpful.

Reply:
Thanks. I believe we need to bring back strong typing to handle polymorphism, I'll circle back.